src/observer/index.js

```diff
@@ -172,7 +173,22 @@ function copyAugment (target, src, keys) {
 
 function defineReactive (obj, key, val) {
   var dep = new Dep()
-  var childOb = Observer.create(val)
+
+  var target = {
+    val: val
+  }
+
+  if (config.convertAllProperties) {
+    var property = Object.getOwnPropertyDescriptor(obj, key)
+    if (property && property.get && property.set) {
+      Object.defineProperty(target, 'val', {
+        get: property.get.bind(obj),
```

Comment:
Since people may use Vue in PhantomJS, I've intentionally avoided using the native `bind`. You can use the `bind` provided by Vue's own util: `_.bind(get, obj)` and remove the bind shim in tests.